OptionSelector/MainTableViewController.m

```diff
@@ -0,0 +1,139 @@
+//
+//  Created by Brian Blanco on 8/14/15.
+//  Copyright © 2015 Mike Kavouras. All rights reserved.
+//
+
+#import "MainTableViewController.h"
+#import "DetailTableViewController.h"
+#import "CQCategory.h"
+
+@interface MainTableViewController ()
+
+@end
+
+@implementation MainTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    CQCategory *invertebratesCategory = [[CQCategory alloc] init];
+    
+    invertebratesCategory.name = @"Invertebrates";
+    invertebratesCategory.options = @[@"Butterfly",
+                                      @"Amoeba",
+                                      @"Fly",
+                                      @"Jellyfish",
+                                      @"Spider",
+                                      @"Beetle"];
+    
+    
+    CQCategory *amphibiansCategory = [[CQCategory alloc] init];
+    
+    amphibiansCategory.name = @"Amphibians";
+    amphibiansCategory.options = @[@"Frog",
+                                   @"Salamander",
+                                   @"Toad",
+                                   @"Mudpuppy",
+                                   @"Eel",
+                                   @"Caecilian"];
+    
+    
+    
+    CQCategory *fishCategory = [[CQCategory alloc] init];
+    
+    fishCategory.name = @"Fish";
+    fishCategory.options = @[@"Catfish",
+                             @"Sea Trout",
+                             @"Sunfish",
+                             @"Salmon",
+                             @"Perch",
+                             @"Halinbut"];
+    
+    
+    CQCategory *reptilesCategory = [[CQCategory alloc] init];
+    
+    reptilesCategory.name = @"Reptiles";
+    reptilesCategory.options = @[@"Snake",
+                                   @"Iguana",
+                                   @"Alligator",
+                                   @"Bearded Dragon",
+                                   @"Savannah Monitor",
+                                   @"Tortoise"];
+    
+    
+    
+    CQCategory *mammalsCategory = [[CQCategory alloc] init];
+    
+    mammalsCategory.name = @"Mammals";
+    mammalsCategory.options = @[@"Cat",
+                                @"Dog",
+                                @"Ferret",
+                                @"Skunk",
+                                @"Rat",
+                                @"Sugar Glider"];
+    
+    
+    
+    
+    CQCategory *birdsCategory = [[CQCategory alloc] init];
+    
+    birdsCategory.name = @"Birds";
+    birdsCategory.options = @[@"African Grey",
+                              @"Cockatiel",
+                              @"Dove",
+                              @"Falcon",
+                              @"Eagle",
+                              @"Sparrow"];
+    
+    self.animalCategories = @[invertebratesCategory, amphibiansCategory, fishCategory, reptilesCategory, mammalsCategory, birdsCategory];
+    
+}
+
+- (void)viewDidAppear:(BOOL)animated {
+    [self.tableView reloadData];
```

Comment:
Two comments here. First, when you're overriding methods like `viewDidLoad` or `viewDidAppear:`, you should always call the `super` implementation first:
```objc
[super viewDidAppear:animated];
//...
```
That way, you'll run the code in the superclass's implementation of this method, and then your code.

Second, you probably noticed that after you select an option in the detail view and then go back, there's a delay before any changes are reflected in the master table view. If you wanted the change to be more instantaneous, you could create a `delegate` property on your detail view controller and set it to `self` when preparing to segue from the master to the detail controller, and send a message to the master controller whenever an option was selected indicating that it should reload its table view.